tests/unit/s2n_random_test.c

```diff
@@ -894,6 +910,48 @@ int main(int argc, char **argv)
 {
     BEGIN_TEST_NO_INIT();
 
+    /* Feature probe: Negative test */
+    {
+        if (s2n_libcrypto_is_awslc()) {
+#if defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE)
+            FAIL_MSG("Expected ENGINE feature probe to be disabled with AWS-LC");
+#endif
+        }
+    };
+
+    /* Feature probe: Positive test
+     *
+     * Expect ENGINE APIs to be available if `OPENSSL_NO_ENGINE` is not
+     * defined.
+     *
+     * TODO: Test is disabled due to unrelated zlib error on AL2.
+     * https://github.com/aws/s2n-tls/issues/4900
```

Comment:
Note I had to disable the positive feature probe test due to a zlib error when compiling on AL2.